docs/root/api/starting_envoy.rst

```diff
@@ -428,6 +428,37 @@ This can help negate the effect of head-of-line (HOL) blocking for slow connecti
   // Swift
   builder.h2ExtendKeepaliveTimeout(true)
 
+
+~~~~~~~~~~~~~~~~~~~~
+``addKeyValueStore``
+~~~~~~~~~~~~~~~~~~~~
+
+Implementations of a public KeyValueStore interface may be added in their respective languages and
+made available to native code. General usage is supported, but typical future usage will be in
+support of HTTP and endpoint property caching.
+
+**Example**::
+
+  // Kotlin
+  builder.addKeyValueStore(MyKeyValueStoreImpl())
```

Comment:
Missing the string parameter.

Maybe we should encourage reverse domain notation to convey that these should be unique?

```suggestion
  builder.addKeyValueStore("com.myapp.envoymobile.MyKeyValueStore", MyKeyValueStoreImpl())
```

Reply:
thanks, good catch